experimental/dds/tree/src/test/HistoryEditFactory.tests.ts

```diff
@@ -50,6 +50,57 @@ describe('revert', () => {
 		expect(revertedChange.source.end.referenceSibling).to.deep.equal(secondNode.identifier);
 	});
 
+	it('handles reverting the insert of empty nodes, with multiple subsequent non-empty nodes', () => {
+		// build and insert of empty traits
+		const emptyTraitNodeId = 0 as DetachedSequenceId;
+		const emptyTraitBuild = ChangeInternal.build([], emptyTraitNodeId);
+		const emptyTraitInsert = ChangeInternal.insert(emptyTraitNodeId, {
+			referenceTrait: testTree.left.traitLocation,
+			side: Side.After,
+		});
+
+		// build and insert of non-empty traits
+		const firstDetachedId = 1 as DetachedSequenceId;
+		const firstNode = testTree.buildLeafInternal();
+		const firstBuild = ChangeInternal.build([firstNode], firstDetachedId);
+		const secondDetachedId = 2 as DetachedSequenceId;
+		const secondNode = testTree.buildLeafInternal();
+		const secondBuild = ChangeInternal.build([secondNode], secondDetachedId);
+		const insertedNodeId = 3 as DetachedSequenceId;
+		const insertedBuild = ChangeInternal.build([firstDetachedId, secondDetachedId], insertedNodeId);
+		const insertChange = ChangeInternal.insert(insertedNodeId, {
+			referenceTrait: testTree.left.traitLocation,
+			side: Side.After,
+		});
+		const result = expectDefined(
+			revert(
+				[emptyTraitBuild, emptyTraitInsert, firstBuild, secondBuild, insertedBuild, insertChange],
+				testTree.view
+			)
+		);
+		expect(result.length).to.equal(1);
+		const revertedChange = result[0] as DetachInternal;
+		expect(revertedChange.source.start.referenceSibling).to.deep.equal(firstNode.identifier);
+		expect(revertedChange.source.end.referenceSibling).to.deep.equal(secondNode.identifier);
+	});
+
+	it('handles reverting the detach of an empty trait', () => {
+		const insertedNodeId = 0 as DetachedSequenceId;
+		const result = revert(
+			[
+				ChangeInternal.detach(
+					StableRangeInternal.all({
+						label: 'someNonExistentTraitLabel' as TraitLabel,
+						parent: testTree.identifier,
+					}),
+					insertedNodeId
+				),
+			],
+			testTree.view
+		);
+		expect(result).to.have.lengthOf(0);
+	});
+
```

Comment:
You could have a test which builds an empty sequence, inserts it, and then inserts it again. Reverting this should return `undefined`. This guards against the code from a day or two ago, which had the bug where we didn't clear the built/detached nodes map (and in that case, it would have succeeded, which would be weird and maybe bad).

Reply:
It might look like an oddly specific test to an uninitiated observer, so feel free to label it with a comment at the top: "this is a regression test for a bug where..." if you like.